keysync/server.go

```diff
@@ -206,9 +228,30 @@ func (ks *KeySyncServer) syncSecretsToLocalKeys(secList *corev1.SecretList, skh
 			path := filepath.Join(ks.keySyncDir, filename)
 			if !fileExists(path) {
 				logrus.Printf("Syncing new key: %v", filename)
-				err := ioutil.WriteFile(path, data, 0600)
+				err := ioutil.WriteFile(path, data, ks.keyFilePermissions)
+				if err != nil {
+					logrus.Errorf("Unable to write file %s: %v", path, err)
+					continue
+				}
+			}
+
+			// Permission and ownership corrigation if needed
```

Comment:
update comment here to indicate that WriteFile does not guarantee that these permissions are set if successful due to umask or existing file, so this is necessary to check